sycl/doc/extensions/proposed/sycl_ext_oneapi_device_specific.asciidoc

```diff
@@ -0,0 +1,1216 @@
+= sycl_ext_oneapi_device_specific
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+// This is necessary for asciidoc, but not for asciidoctor
+:cpp: C++
+:dpcpp: DPC++
+
+== Introduction
+In OpenCL 2.0 and later, a user is able to allocate program
+scope memory which can be accessed like a {cpp} global variable by any kernel in
+an OpenCL program (`cl_program`). When a program is shared between multiple
+devices, each device receives its own unique instance of the program scope
+memory allocation.
+
+This extension introduces device-specific memory allocations into SYCL that can be
+accessed within a kernel using syntax similar to {cpp} global variables, but
+that have unique instances per `sycl::device` and `sycl::context`. Mechanisms
+are provided for the host program to enqueue copies to or from device-specific
+allocations.  Restrictions are placed on the types of data that can be
+stored within `device_specific` allocations, particularly around copyability and
+constructors/destructors.
+
+== Notice
+
+Copyright (c) 2021 - 2023 Intel Corporation.  All rights reserved.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are
+trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc.
+used by permission by Khronos.
+
+== Status
+
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
```

Comment:
Let's update this to the suggested text according to the [template](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/template.asciidoc#status):

```
This is a proposed update to an existing extension.  Interfaces defined in this
specification may not be implemented yet or may be in a preliminary state.  The
specification itself may also change in incompatible ways before it is
finalized.  *Shipping software products should not rely on APIs defined in this
specification.*  See
link:../supported/sycl_ext_oneapi_myextension.asciidoc[here] for the existing
extension, which is implemented.
```

Reply:
Fixed in ec7097a6dd34c0d9b6401f94b197f5da80d7b829.